### PR TITLE
App Engine, Upgrade Pathway, Python3

### PR DESCRIPTION
**Pull request by thinkocapo**
## Done
- Fixed edge case due to timeclock skew between Dev and Prod that happened around last day of the month (affected CalVer Release versioning)
```
npm run build
...
"build": "REACT_APP_RELEASE=`../calver.sh` react-scripts build",
```

- Upgrade pathway outlined for everyone to follow. Full `git`, `npm`, `pip` commands documented.

## Python3
Is now used, and required. Exception thrown and system.exit() if you run on Python2.

Here's the new docker image we're using for Cloud Run.
https://hub.docker.com/_/python `3.8-slim`

app.yaml (AppEngine) specifies its own as `python38`

## Versions
Updated flask, flask-cors, gunicorn

## App Engine
The Cloud Run workflow was preserved (untouched) but AppEngine support has been added. Still run development mode by 'make all' in /flask/Makefile.

Update your react/.env with appspot (Appengine) URL's and:
```
cd flask && gcloud app deploy
cd react && gcloud app deploy
```

App Engine replaces the following:
```
flask/Dockerfile
cloudbuild.yaml
gcloud builds submit
gcloud run deploy
```
with:
App.yaml, 2 lines of code and `gcloud app deploy` :)
```
service: tracing-flask
runtime: python38
```

Used `venv` instead of `virtualenv`.

`app.py` renamed `main.py` because PaaS products look for `main.py` for entrypoint and we're not using a Dockerfile (can specify entrypoint)

### CODE REVIEW

File: flask/main.py [hunk 1]
```diff
@@ -78,7 +75,7 @@ def process_order(cart):
             raise Exception("Not enough inventory for " + item['type'])
         else:
             tempInventory[item['type']] -= 1
-            print 'Success: ' + item['type'] + ' was purchased, remaining stock is ' + str(tempInventory[item['type']])
```
Comment (thinkocapo) [inline flask/main.py:81]: Python2 syntax

File: react/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "deploy": "serve -s build",
     "deploylocal": "npm run buildlocal && make setup_release && serve -s build",
     "start": "REACT_APP_RELEASE=`../calver.sh` react-scripts start",
-    "build": "react-scripts build",
+    "build": "REACT_APP_RELEASE=`../calver.sh` react-scripts build",
```
Comment (thinkocapo) [inline react/package.json:34]: @dgbailey this should fix the problem you were having with your Release discrepancy between what sentry-cli was doing, and what the React javascript code was doing.

You can test it locally too:
```
npm run build
serve -s build
```
^ then put a log where it checks to see if REACT_APP_RELEASE was defined...
```
if (process.env.REACT_APP_RELEASE == null) {
    console.log('release is null....')
```
^ it should not enter that execution block anymore.

File: flask/main.py [hunk 3]
```diff
@@ -93,7 +90,7 @@ def sentry_event_context():
 def checkout():
 
     order = json.loads(request.data)
-    print "Processing order for: " + request.headers.get('email')
```
Comment (thinkocapo) [inline flask/main.py:96]: Python2

File: flask/Makefile [hunk 4]
```diff
@@ -1,2 +1,2 @@
 deploy:
-	FLASK_APP=app.py FLASK_ENV='test' flask run -p 3003
```
Comment (thinkocapo) [inline flask/Makefile:2]: `main` is the expected entry point in a lot of cloud products (e.g. AWS Lambdas look for 'main' executable, and App Engine Python Standard Environment looks for 'main.py')

so let's stick with convention.

File: react/package.json (hunk 2 above)
Comment (dgbailey) [reply react/package.json:34]: Hello @thinkocapo, was able to deploy this on GCP and verify that the discrepancy between the release used by the cli and init() statement is resolved.

**Pull request merged**